### PR TITLE
Ensure pixel_ratio is applied when computing resampling coordinates

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/holoviews/issues/6372

Before:

https://github.com/user-attachments/assets/d5dc6738-d6f8-4b88-9355-fb073359b6b7

After:

https://github.com/user-attachments/assets/639a6b70-ed25-41f5-8642-3a4bcd29db31

```python
import pandas as pd
import holoviews as hv
from holoviews.operation import apply_when
from holoviews.operation.datashader import rasterize
from functools import partial
import panel as pn

hv.extension("bokeh")

df = pd.DataFrame(
    {
        "lon": [45.5531, 46.4731, 47.4731, 48.41234],
        "lat": [34.5531, 35.4731, 36.4731, 37.41234],
    }
)

def _resample_obj(operation, obj, opts):
    def exceeds_resample_when(plot):
        return len(plot) > 1000

    processed = apply_when(
        obj, operation=partial(operation, **opts), predicate=exceeds_resample_when
    )
    return processed


points = hv.Points(df, ["lon", "lat"])
resampled_points = _resample_obj(rasterize, points, {})
pn.serve(resampled_points)
```

### CODE REVIEW

File: holoviews/operation/datashader.py [hunk 1]
```diff
@@ -362,7 +362,7 @@ def _process(self, element, key=None):
 
         if self.p.precompute:
             self._precomputed[element._plot_id] = x, y, data, glyph
-        (x_range, y_range), (xs, ys), (width, height), (xtype, ytype) = self._get_sampling(element, x, y)
+        (x_range, y_range), (xs, ys), (width, height), (xtype, ytype) = self._get_sampling(element, x, y, apply_pixel_ratio=False)
```
Comment (ahuang11) [inline holoviews/operation/datashader.py:365]: Without this, it'll error below on line 374 because width/height mismatch len(xs) and len(ys)

Comment (philippjfr) [reply holoviews/operation/datashader.py:365]: This defeats the point of the pixel ratio entirely as far as I can tell.

File: holoviews/tests/ui/bokeh/test_datashader.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+except ImportError:
+    raise SkipTest("Datashader not available")
+from ...plotting.utils import ParamLogStream
+from .. import expect, wait_until
+
+pytestmark = pytest.mark.ui
+
+
+@pytest.mark.usefixtures("bokeh_backend")
+def test_apply_when_zoom(serve_hv):
```
Comment (philippjfr) [inline holoviews/tests/ui/bokeh/test_datashader.py:21]: I don't know why this would ever have to be a UI test.

Comment (philippjfr) [reply holoviews/tests/ui/bokeh/test_datashader.py:21]: UI tests should only be implemented for functionality that actually happens on the frontend. In this case you can simulate all the conditions entirely in Python.

File: holoviews/operation/datashader.py (hunk 1 above)
Comment (philippjfr) [reply holoviews/operation/datashader.py:365]: Yeah, the correct fix here is to multiply the width/height before creating the coordinates, not to simply disable pixel ratio entirely.

**Review by philippjfr — APPROVED** [review]

**Pull request merged**